checks.md

```diff
@@ -161,3 +161,12 @@ This check tries to determine if the project has branch protection enabled.
   force pushes or deletion of your important branches.
 - For GitHub, check out the steps
   [here](https://docs.github.com/en/github/administering-a-repository/managing-a-branch-protection-rule).
+
+
+## Automatic-Dependency-Update
+
+This check tries to determine if the project has dependencies automatically updated.
+
+**Remediation steps**:
+- Enable dependabot for the project and have a dependabot.yml file in the repository. 
+ 
```

Comment:
This is an amazing check, but there are two problems:
1. Dependabot can be enabled on org level
2. Dependabot can be enabled via UI
and this is substantial portion of things, and we would show incorrect result for these. maybe we should bring some of these problems with github api in next openssf sync, e.g. branch protection, this one with knowing security settings like dependabot, etc. Landing this as-is will be problematic since we cannot get right status in many cases and scorecard needs to be accurate. thoughts?

Reply:
> This is an amazing check, but there are two problems:
> 
> 1. Dependabot can be enabled on org level
> 2. Dependabot can be enabled via UI
>    and this is substantial portion of things, and we would show incorrect result for these. maybe we should bring some of these problems with github api in next openssf sync, e.g. branch protection, this one with knowing security settings like dependabot, etc. Landing this as-is will be problematic since we cannot get right status in many cases and scorecard needs to be accurate. thoughts?

I agree this might be an issue. We can have in-depth discussion. 

Also

- But why not provide an results with lesser confidence, we can never be 100% in some of the cases.

- Another way to solve this issues is to look for commits and check for `Author: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>` and check for signed commits

Reply:
yes can be then

- look for that file .github/dependabot.yml, if exists, pass with high confidence.
- if failed, check for commits [we have to see how far we can go without burning quota, or instead see if we can get user dependabot events by api]. if that exists, pass with medium-high confidence, depending on last update?
- otherwise fail with low confidence (like 3?)

thoughts, can also brainstorm in next meeting with @azeemshaikh38 . but even this is a good start i think.

Reply:
Sounds good.

Reply:
Some projects may have dependabot for tooling (let's say a primary C++ project with no dependabot, but has python deps that utilize dependabot). 

I wonder if we should lower confidence for projects with unsupported languages

Reply:
re: dependabot commits. If people have dependabot installed, we might still want to verify that they accept dependabot's PRs once in a while.

Reply:
yes right, this might be a better or definitely needed part here, which is did any dependendabot cls land in a while.